### DISCUSSION TITLE
Some simple questions concerning Voila

### DISCUSSION THREAD

**Discussion by Jacques2101** [voila-dashboards/voila#1455] [Q&A] [answered]
### Question

I have some questions concerning Voila. I ma new to this app. 

1) Does it work with VSCode ? 
2) Does it produce a standalone app ? 
3) what is the difference with ipywidgets ?

Thx

**Comment by martinRenou** [comment, marked as answer]
> 1. Does it work with VSCode ?

AFAIK VSCode does not have a voila-preview plugin.

> 2. Does it produce a standalone app ?

Not sure what you mean by that. Voila, like Jupyter Notebook or JupyterLab start a jupyter server to serve and render notebooks. The difference is that it executes the notebook entirely and renders the outputs only by default. Also notebooks are readonly with Voila.

> 3. what is the difference with ipywidgets ?

IPywidgets is a Python library to create interactive widgets in your Notebook. These widgets will work in Voila, Jupyter Notebook or JupyterLab.

**Comment by Jacques2101** [comment]
I would like to make a dashboard with code hidden and with widgets (to select some options from my code). 
Is Voila the right place to be ? Is it more ipywidgets ? 

Thx a lot for your answer

PS : I am sorry but i know nothing about Voila

**Comment by martinRenou** [comment]
Sounds like Voila is exactly what you're looking for :)

**Comment by Jacques2101** [comment]
and, last but not the least question, what is the difference with streamlit ?

**Comment by martinRenou** [comment]
Streamlit is not notebook-oriented. Voila is puts the Jupyter notebook at its core.

You basically use the Jupyter Notebook or JupyterLab for the exploratory phase, then execute the resulting notebook with Voila to make it a nice dashboard.

**Reply by Jacques2101** [reply]
Did the dashboard after that needs a python to be installed to be executed. I mean if I want to share it

**Reply by trungleduc** [reply]
yes, you need to have Voila installed to be able to execute your notebook as a dashboard.

**Comment by trungleduc** [comment]
Moving this thread to the discussion section for more visibility.
